addons/sale/models/payment_transaction.py

```diff
@@ -89,36 +84,39 @@ def _log_message_on_linked_documents(self, message):
 
     def _reconcile_after_done(self):
         """ Override of payment to automatically confirm quotations and generate invoices. """
-        sales_orders = self.mapped('sale_order_ids').filtered(lambda so: so.state in ('draft', 'sent'))
-        for tx in self:
-            tx._check_amount_and_confirm_order()
+        self._check_amount_and_confirm_order()
         # send order confirmation mail
-        sales_orders._send_order_confirmation_mail()
-        # invoice the sale orders if needed
-        self._invoice_sale_orders()
-        res = super()._reconcile_after_done()
-        if self.env['ir.config_parameter'].sudo().get_param('sale.automatic_invoice') and any(so.state in ('sale', 'done') for so in self.sale_order_ids):
-            default_template = self.env['ir.config_parameter'].sudo().get_param('sale.default_invoice_email_template')
-            if default_template:
-                for trans in self.filtered(lambda t: t.sale_order_ids.filtered(lambda so: so.state in ('sale', 'done'))):
-                    trans = trans.with_company(trans.acquirer_id.company_id).with_context(
-                        mark_invoice_as_sent=True,
-                        company_id=trans.acquirer_id.company_id.id,
-                    )
-                    for invoice in trans.invoice_ids.with_user(SUPERUSER_ID):
-                        invoice.message_post_with_template(int(default_template), email_layout_xmlid="mail.mail_notification_paynow")
-        return res
+        self.mapped('sale_order_ids').filtered(
```

Comment:
```suggestion
        self.sale_order_ids.filtered(
```
mapped is not necessary for relational fields